### ISSUE TITLE
`hyp3`: more metadata for `HDF-EOS5` support

### ISSUE THREAD

**Issue by yunjunz** [insarlab/MintPy#1162]
**Description of the desired feature**

[`HDF-EOS5` file format](https://mintpy.readthedocs.io/en/latest/hdfeos5/) is an easy way to save the final time-series product into a single file. There is now near-automatic support for that after https://github.com/insarlab/MintPy/pull/1161. However, the following metadata is missing. It will be great to provide them from the ASF HyP3 side:
   - relativeOrbit: required by HDF-EOS5
   - startUTC: used to calculate the first frame number
   - stopUTC: used to calculate the last frame number
   - ascendingNoteTime: use to calculate the first/last frame number [updated]

Then, all hyp3 products can be saved in HDF-EOS5 format, if the user just turn on the `mintpy.save.hdfEos5` option, without any manual adding of metadata (as I did in [here](https://github.com/yunjunz/MintPy/blob/51fe70e6e85fc62620160730be96ceff82e32053/tests/configs/RidgecrestSenDT71.txt#L26)).

Let me know what you think please. @forrestfwilliams @talogan @jhkennedy 

Next, we would need to update the [example ASF HyP3 dataset](https://github.com/insarlab/MintPy/blob/main/docs/demo_dataset.md#sentinel-1-of-the-2019-ridgecrest-california-earthquake-sequence-with-hyp3) (to include the above metadata); and cut for a new release of mintpy.

**Comment by jhkennedy** [comment]
@yunjunz I agree that supporting HDF-EOS5 files is a great idea -- we're working on a few things on the HyP3 side to provide better metadata overall and effectively eliminate the need of the `prep_hyp3` code here and any hyp3-specific tutorials. There's no timeline, but I'll keep you posted as things mature on our end.

**Comment by forrestfwilliams** [comment]
@yunjunz sorry for the slow reply. As @jhkennedy mentioned, we're working on some things to ease the transition from hyp3 -> mintpy. In the meantime however, yep we can add those three attributes to `prep_hyp3`. start/stop UTC will be simple, but relativeOrbit will take tiny a bit of work. We only provide absolute orbits in the hyp3 metadata so we'll need to convert between the two, but that's a simple calculation. I'll open a PR next week!

**Comment by yunjunz** [comment]
Reopen the issue as the `ascendingNodeTime` is still missing, thus, could not calculate the first/last_frame number yet.

**Comment by forrestfwilliams** [comment]
Hey @yunjunz, the `ascendingNodeTime` will be more work to add to our HyP3 products. We currently don't include this field in our parameter files and will need to update both of our insar processing workflows to accommodate this. Do you have documentation on how the `ascendingNodeTime` should be specified for HDF-EOS?

I'll let you know when our team starts in on this work, but right now I'm not sure when that will be.

**Comment by yunjunz** [comment]
What HDF-EOS5 need is the `first_frame` and `last_frame`, as defined by ASF, which can be calculated as below using `startUTC`, `stopUTC` and `ascendingNodeTime `:
https://github.com/insarlab/MintPy/blob/4d34412b811016e02b9dfc45dc6af519bca85f6a/src/mintpy/utils/isce_utils.py#L203-L205
